packages/celotool/src/lib/prometheus.ts

```diff
@@ -47,30 +51,48 @@ export async function removeHelmRelease() {
   await removeGenericHelmChart(releaseName)
 }
 
-async function helmParameters(clusterConfig: AzureClusterConfig) {
-  return [
+export async function upgradePrometheus() {
+  await createNamespaceIfNotExists(kubeNamespace)
+  return upgradeGenericHelmChart(kubeNamespace, releaseName, helmChartPath, await helmParameters())
+}
+
+async function helmParameters(clusterConfig?: AzureClusterConfig) {
+  const params = [
     `--set namespace=${kubeNamespace}`,
-    `--set cluster=${clusterConfig.clusterName}`,
     `--set gcloud.project=${fetchEnv(envVar.TESTNET_PROJECT_NAME)}`,
     `--set gcloud.region=${fetchEnv(envVar.KUBERNETES_CLUSTER_ZONE)}`,
     `--set sidecar.imageTag=${sidecarImageTag}`,
     `--set prometheus.imageTag=${prometheusImageTag}`,
-    `--set gcloudServiceAccountKeyBase64=${await getPrometheusGcloudServiceAccountKeyBase64(
-      clusterConfig.clusterName
-    )}`,
+    `--set stackdriver_metrics_prefix=${prometheusImageTag}`,
     // Stackdriver allows a maximum of 10 custom labels. kube-state-metrics
     // has some metrics of the form "kube_.+_labels" that provides the labels
     // of k8s resources as metric labels. If some k8s resources have too many labels,
     // this results in a bunch of errors when the sidecar tries to send metrics to Stackdriver.
     `--set-string includeFilter='\\{job=~".+"\\,__name__!~"kube_.+_labels"\\}'`,
   ]
+  if (clusterConfig) {
+    params.push(
+      `--set cluster=${clusterConfig.clusterName}`,
+      `--set stackdriver_metrics_prefix=external.googleapis.com/prometheus/${clusterConfig.clusterName}`,
+      `--set gcloudServiceAccountKeyBase64=${await getPrometheusGcloudServiceAccountKeyBase64forAKS(
+        clusterConfig.clusterName
+      )}`
+    )
+  } else {
+    const clusterName = fetchEnv(envVar.KUBERNETES_CLUSTER_NAME)
+    params.push(
+      `--set cluster=${clusterName}`,
+      `--set stackdriver_metrics_prefix=external.googleapis.com/prometheus/${clusterName}`
```

Comment:
Why are we setting the cluster as part of the prefix, shouldn't the cluster_name be a label of the metric itself?

Reply:
So with current prometheus config the cluster_name is not a label, but we can it if we find that approach better.

Reply:
I guess I'm not exactly sure what is deployed where, but at least the eksportisto metrics have the `cluster_name` label and they seem correct

Reply:
I guess it was added by eksportio. I did not see that label for other metrics. I added it as an extra label

Reply:
I guess it was added by eksportisto itself. Added as an extra label for all metrics